lib/linguist/generated.rb

```diff
@@ -588,7 +588,8 @@ def generated_graphql_relay?
     def generated_gamemakerstudio?
       return false unless ['.yy', '.yyp'].include? extname
       return false unless lines.count > 3
-      return lines[2].match(/\"modelName\"\:\s*\"GM/)
+      return lines[2].match(/\"modelName\"\:\s*\"GM/) ||
+             lines[0].start_with?("1.0.0")
```

Comment:
Would there be a way for me to include the weird character back in there? Its this one: ←(``U+2190`` - ``0xE2 0x86 0x90`` in utf-8).

It might make the checking a lot stricter, and then could possible replace starts with ``1.0.0`` with ``\d\.\d\.\d←`` which would then support future versions :)

Reply:
PS: Just throwing the character in there breaks the testsuite;
```
 1) Error:
TestBlob#test_language:
Encoding::CompatibilityError: incompatible character encodings: ASCII-8BIT and UTF-8
    /home/travis/build/github/linguist/lib/linguist/generated.rb:592:in `start_with?'
    /home/travis/build/github/linguist/lib/linguist/generated.rb:592:in `generated_gamemakerstudio?'
    /home/travis/build/github/linguist/lib/linguist/generated.rb:93:in `generated?'
    /home/travis/build/github/linguist/lib/linguist/generated.rb:12:in `generated?'
    /home/travis/build/github/linguist/lib/linguist/blob_helper.rb:362:in `generated?'
    /home/travis/build/github/linguist/test/test_blob.rb:281:in `block (2 levels) in test_language'
    /home/travis/build/github/linguist/test/test_blob.rb:271:in `each'
    /home/travis/build/github/linguist/test/test_blob.rb:271:in `block in test_language'
    /home/travis/build/github/linguist/test/test_blob.rb:265:in `each'
    /home/travis/build/github/linguist/test/test_blob.rb:265:in `test_language'
  2) Error:
TestFileBlob#test_language:
Encoding::CompatibilityError: incompatible character encodings: ASCII-8BIT and UTF-8
    /home/travis/build/github/linguist/lib/linguist/generated.rb:592:in `start_with?'
    /home/travis/build/github/linguist/lib/linguist/generated.rb:592:in `generated_gamemakerstudio?'
    /home/travis/build/github/linguist/lib/linguist/generated.rb:93:in `generated?'
    /home/travis/build/github/linguist/lib/linguist/generated.rb:12:in `generated?'
    /home/travis/build/github/linguist/lib/linguist/blob_helper.rb:362:in `generated?'
    /home/travis/build/github/linguist/test/test_file_blob.rb:676:in `block (2 levels) in test_language'
    /home/travis/build/github/linguist/test/test_file_blob.rb:666:in `each'
    /home/travis/build/github/linguist/test/test_file_blob.rb:666:in `block in test_language'
    /home/travis/build/github/linguist/test/test_file_blob.rb:660:in `each'
    /home/travis/build/github/linguist/test/test_file_blob.rb:660:in `test_language'
```

Reply:
@RobQuistNL Is `LANG` set in your environment? On OpenBSD, every test belched spurious encoding errors until I executed tests using `LANG=en_AU.UTF-8 bundle exec rake` (aliased in my shell as simply `brake`).

It fixed errors which looked extremely similar to the ones you've posted.

Reply:
And it was fixed by removing the ``←`` character from the matching line

Reply:
I must note that I have not ran the suite locally - I tried it a while back but I feel like I'm too unexperienced with ruby / whatnot and am missing a lot of tooling ^_^ Full-blown PHP developer here

Reply:
Not an expert on how encodings work in Ruby (especially in CI), but you can express Unicode characters as 7-bit ASCII using:

~~~rb
/\u2190/ # RegExp escape
"\u2190" # Analogous string escape

"\u2190" =~ /\u2190/ # => Returns 0, the index of the first match
"foobar" =~ /\u2190/ # => nil No match
~~~

Reply:
The `=~` operator is ~~shameless ripped off from Perl~~ Ruby's way of performing a match between a regular expression and a value.

In PHP, the equivalent syntax is `pcre_match("/\\u2190/", "input");` (if memory serves, it's been years since I bothered working with the language. #NODE4LYF

Reply:
Ah cool, will try :)

Reply:
I've looked into this today and attempting to match the `←` is definitely a bit of a struggle. I can get it working locally on macOS by using that char directly, but it doesn't play nice on Travis.

An alternate solution may be to more closely match the format of the line ignoring the content, for example:

```
lines[0] =~ /^\d\.\d\.\d.+\|\{/
```

... matches the `1.0.0` at the beginning of the line and the `|{` at the end, ignoring everything in between and does the trick locally and on [Travis here](https://travis-ci.org/github/linguist/builds/449346384). We could probably get more fancy with the regex, but I'm not sure it's worth the effort.

Reply:
Ah thanks, I didn't see your comment there for a while. Sorry! Yeah, I think we're going to have to walk this way since the actual char is not going to work.